src/processes/atmos/atmosphere.jl

```diff
@@ -0,0 +1,123 @@
+    NF,
+    tracervars,
+    Grid<:AbstractLandGrid{NF},
+    AirTemp,
+    Humidity,
+    Pressure,
+    Windspeed<:NTuple{2},
+    Precip<:AbstractPrecipitation,
+    SolarRad<:AbstractSolarRadiation,
+    Tracers<:NamedTuple{tracervars,<:Tuple{Vararg{TracerGas}}},
+} <: AbstractBoundaryConditions
+    "Spatial grid"
+    grid::Grid
+
+    "Surface-relative altitude in meters at which the atmospheric forcings are assumed to be applied"
+    altiude::NF = 2.0
+    
+    "Near-surface air temperature in °C"
+    T_air::AirTemp = nothing
+
+    "Near-surface specific humidity in kg/kg"
+    humidity::Humidity = nothing
+
+    "Near-surface atmospheric pressure in Pa"
+    pressure::Pressure = nothing
+
+    "Windspeed velocity components in m/s"
+    windspeed::NamedTuple{(:u,:v), Windspeed} = (u=nothing, v=nothing)
+
+    "Precipitation scheme"
+    precip::Precip = TwoPhasePrecipitation()
+
+    "Downwelling solar radiation scheme"
+    solar::SolarRad = TwoBandSolarRadiation()
+
+    "Atmospheric tracer gases"
+    tracers::Tracers = default_tracers(grid)
+end
+
+variables(atm::AtmosphericState) = (
+    auxiliary(:T_air, XY(), units=u"°C", desc="Air temperature"),
+    auxiliary(:humidity, XY(), units=u"kg/kg", desc="Specific humidity"),
+    auxiliary(:pres, XY(), units=u"Pa", desc="Atmospheric pressure at the surface"),
+    auxiliary(:wind_u, XY(), units="m/s", desc="Wind speed u-component"),
+    auxiliary(:wind_v, XY(), units="m/s", desc="Wind speed v-component"),
+    variables(atm.precip)...,
+    variables(atm.solar)...,
+    # splat all tracer variables into tuple
+    tuplejoin(map(variables, atm.tracers)...)...,
+)
+
+function compute_auxiliary!(state, model, atm::AtmosphericState)
+    set!(state.T_air, atm.T_air)
+    set!(state.humidity, atm.humidity)
+    set!(state.pres, atm.pres)
+    set!(state.wind_u, atm.wind_u)
+    set!(state.wind_v, atm.wind_v)
+    compute_auxiliary!(state, model, atm.precip)
+    compute_auxiliary!(state, model, atm.solar)
+    for tracer in atm.tracers
+        compute_auxiliary!(state, model, tracer)
+    end
+end
+
+@kwdef struct TwoPhasePrecipitation{RF, SF} <: AbstractPrecipitation
+    rainfall::RF = nothing
+    snowfall::SF = nothing
+end
+
+variables(::TwoPhasePrecipitation) = (
+    auxiliary(:rainfall, XY(), units=u"m/s", desc="Liquid precipitation (rainfall) rate"),
+    auxiliary(:snowfall, XY(), units=u"m/s", desc="Frozen precipitation (snowfall) rate"),
+)
+
+function compute_auxiliary!(state, model, precip::TwoPhasePrecipitation)
+    set!(state.rainfall, precip.rainfall)
+    set!(state.snowfall, precip.snowfall)
+end
+
+@kwdef struct TwoBandSolarRadiation{SW, LW} <: AbstractSolarRadiation
+    shortwave::SW = nothing
+    longwave::LW = nothing
+end
+
+variables(::TwoBandSolarRadiation) = (
+    auxiliary(:SwIn, XY(), units=u"W/m^2", desc="Incoming (downwelling) shortwave radiation"),
+    auxiliary(:LwIn, XY(), units=u"W/m^2", desc="Incoming (downwelling) longwave radiation"),
+)
```

Comment:
Your idea is to use shortwave and longwave radiation as forcing but the upwelling fluxes would be computed in the radiation scheme? Then land would need to alter somehow the albedo the atmosphere sees and communicate its skin temperature so that the radiation scheme knows its bottom boundary condition. Sounds good, just asking for clarification